build/e2e_test.go

```diff
@@ -153,6 +158,88 @@ func TestBasic(t *testing.T) {
 	})
 }
 
+func TestBasicTenant(t *testing.T) {
```

Comment:
minor: should this test live inside of the tenant package?

Reply:
That would create circular imports. Why do you want to move it? To not spill too much "tenant" into the codebase? We could put it in a separate package inside internal/tenant/